packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -46,11 +98,34 @@ - (instancetype)initWithQueue:(nonnull SKPaymentQueue *)queue
 }
 
 - (void)startObservingPaymentQueue {
-  [_queue addTransactionObserver:self];
+  self.isObservingTransactions = true;
+
+  [self processCachedTransactions];
 }
 
 - (void)stopObservingPaymentQueue {
-  [_queue removeTransactionObserver:self];
+  self.isObservingTransactions = false;
+}
+
+- (void)processCachedTransactions {
+  NSArray *cachedObjects =
+      [self.transactionCache getObjectsForKey:TransactionCacheKeyUpdatedTransactions];
+  if (cachedObjects) {
+    self.transactionsUpdated(cachedObjects);
+    [self.transactionCache removeObjectsForKey:TransactionCacheKeyUpdatedTransactions];
+  }
+
+  cachedObjects = [self.transactionCache getObjectsForKey:TransactionCacheKeyUpdatedDownloads];
+  if (cachedObjects) {
+    self.updatedDownloads(cachedObjects);
+    [self.transactionCache removeObjectsForKey:TransactionCacheKeyUpdatedDownloads];
+  }
+
+  cachedObjects = [self.transactionCache getObjectsForKey:TransactionCacheKeyRemovedTransactions];
+  if (cachedObjects) {
+    self.transactionsRemoved(cachedObjects);
+    [self.transactionCache removeObjectsForKey:TransactionCacheKeyRemovedTransactions];
+  }
```

Comment:
If the only use case for removing things from the cache is clearing the entire cache, why not replace `removeObjectsForKey` with `clear`?

Reply:
You are right, the only functionality we need is to clear the whole cache. Refactored the implementation to remove all objects and renamed the method to `clear`.